api/types/github.go

```diff
@@ -282,6 +323,33 @@ func (c *GithubConnectorV3) MapClaims(claims GithubClaims) ([]string, []string,
 	return utils.Deduplicate(roles), utils.Deduplicate(kubeGroups), utils.Deduplicate(kubeUsers)
 }
 
+// checkOrgExternalSSO returns true if the Github organization is a Github
+// Enterprise Cloud organization that has external SSO enabled. SSO is a
+// Teleport Enterprise feature so it should not be allowed for OSS.
+func checkOrgExternalSSO(org string) (bool, error) {
+	// A Github organization will have a "sso" page reachable if it
+	// supports external SSO. There doesn't seem to be any way to get this
+	// information from the Github REST API without being an owner of the
+	// Github organization, so check if this exists instead.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
nit: we should probably use our default HTTP timeout here https://github.com/gravitational/teleport/blob/0e937edd4d841d33240a9838f2c592a8e9626baf/lib/defaults/defaults.go#L116